src/test/java/io/jenkins/plugins/coverage/CoveragePublisherPipelineTest.java

```diff
@@ -0,0 +1,169 @@
+
+        private List<Threshold> globalThresholds = new LinkedList<>();
+
+        private CoverageScriptedPipelineScriptBuilder() {
+        }
+
+
+        public static CoverageScriptedPipelineScriptBuilder builder() {
+            return new CoverageScriptedPipelineScriptBuilder();
+        }
+
+        public String getAutoDetectPath() {
+            return autoDetectPath;
+        }
+
+        public CoverageScriptedPipelineScriptBuilder setAutoDetectPath(String autoDetectPath) {
+            this.autoDetectPath = autoDetectPath;
+            return this;
+        }
+
+        public CoverageScriptedPipelineScriptBuilder addAdapter(CoverageReportAdapter adapter) {
+            adapters.add(adapter);
+            return this;
+        }
+
+        public CoverageScriptedPipelineScriptBuilder addGlobalThreshold(Threshold threshold) {
+            globalThresholds.add(threshold);
+            return this;
+        }
+
+
+        public String build() {
+            StringBuilder sb = new StringBuilder();
+            sb.append("node {")
+                    .append("publishCoverage(");
+
+            sb.append("autoDetectPath:'").append(autoDetectPath).append("'");
+
+            if (adapters.size() > 0) {
+                sb.append(",adapters:[");
+                for (int i = 0; i < adapters.size(); i++) {
+                    if (i > 0) sb.append(",");
+
+                    CoverageReportAdapter adapter = adapters.get(i);
+                    sb.append(generateSnippetForAdapter(adapter));
+                }
+                sb.append(']');
+            }
+
+            if (globalThresholds.size() > 0) {
+                sb.append(",globalThresholds: [");
+                for (int i = 0; i < globalThresholds.size(); i++) {
+                    if (i > 0) sb.append(",");
+
+                    Threshold threshold = globalThresholds.get(i);
+                    sb.append(generateSnippetForThreshold(threshold));
+                }
+                sb.append("]");
+            }
+
+            sb.append(")")
+                    .append("}");
+            return sb.toString();
+        }
+
+        private String generateSnippetForAdapter(CoverageReportAdapter adapter) {
+            Descriptor<CoverageReportAdapter> d = adapter.getDescriptor();
+            Class c = d.getClass();
+
+            if (c.isAnnotationPresent(Symbol.class)) {
+                Symbol s = (Symbol) c.getAnnotation(Symbol.class);
+                String[] symbolValues = s.value();
+
+                if (symbolValues.length > 0) {
+                    String symbol = symbolValues[0];
+
+                    return symbol + "(path:'" + adapter.getPath() + "')";
+                }
+            }
+
+            //TODO condition when adapter do not have @Symbol annotation.
+            return "";
+        }
+
+        private String generateSnippetForThreshold(Threshold threshold) {
+
+            return "[healthyThresh:" + threshold.getHealthyThresh()
```

Comment:
"thresh" is probably a bad wording for UI and API. It has different meaning (@jeffpearce please correct me if I am wrong). Does it come from Cobertura plugin? If no, I would recommend using a full "threashold" everywhere in API and Symbols

Reply:
Yeah, I will change them in threshold branch.